core/rest-client/src/http_client.rs

```diff
@@ -56,19 +55,76 @@ pub trait SendHttpRequest {
 		T: RestPath<U>;
 }
 
+/// Send trait used by the http client to send HTTP request, based on `http_req`.
+pub trait Send: Default {
+	fn execute_send_request(
+		&self,
+		request: &mut Request,
+		writer: &mut Vec<u8>,
+	) -> Result<Response, Error>;
+}
+
 /// HTTP client implementation
 ///
 /// wrapper for the `http_req` library that adds the necessary headers and body to a request
-pub struct HttpClient {
+pub struct HttpClient<SendType> {
+	send: SendType,
 	send_null_body: bool,
 	timeout: Option<Duration>,
 	headers: Headers,
 	authorization: Option<String>,
 }
 
-impl Default for HttpClient {
+/// Default send method.
+/// Automatically upgrades to TLS in case the base URL contains 'https'
+/// For https requests, the default trusted server's certificates
+/// are provided by the default tls configuration of the http_req lib
+#[derive(Default)]
+pub struct DefaultSend;
+
+impl Send for DefaultSend {
+	fn execute_send_request(
+		&self,
+		request: &mut Request,
+		writer: &mut Vec<u8>,
+	) -> Result<Response, Error> {
+		request.send(writer).map_err(Error::HttpReqError)
+	}
+}
+
+/// Sends a HTTPs request with the server's root certificate.
+/// The connection will only be established if the supplied certificate
+/// matches the server's root certificate.
+#[derive(Default)]
+pub struct SendWithCertificateVerification {
+	root_certificate: String,
+}
```

Comment:
So when this is default constructed, the `String` will be empty. And the you pass a `Some()` with an empty string to the `send_with_pem_certificate`. So if you really want the default trait here, I'd suggest the root certificate to be an option: `maybe_root_certificate: Option<String>`. Otherwise, if you always want a certificate to be present for this implementation, we cannot have `Default`.